### PR TITLE
Enhance the room account data API naming

### PR DESCRIPTION
**Pull request by giomfo**
API Changes:
 - RoomAccountdata.hasTags() has been deprecated. Use .hasRoomTags() instead.
 - RoomAccountdata.getKeys() has been deprecated. Use .getRoomTagsKeys() instead.
 - RoomAccountdata.handleTagEvent() has been removed. Use .handleEvent() instead.
 - IMXStore.setRoomsWithoutURLPreview() has been removed.
 - IMXStore.getRoomsWithoutURLPreviews() has been removed. Use RoomAccountdata.isURLPreviewAllowedByUser() instead.

### CODE REVIEW

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java [hunk 1]
```diff
@@ -1971,26 +1969,13 @@ private void handleRoomAccountDataEvents(List<Event> accountDataEvents) {
                         }
                     }
                 } else {
-                    mAccountData.handleTagEvent(accountDataEvent);
+                    mAccountData.handleEvent(accountDataEvent);
                     if (Event.EVENT_TYPE_TAGS.equals(accountDataEvent.getType())) {
                         if (summary != null) {
-                            summary.setRoomTags(mAccountData.getKeys());
+                            summary.setRoomTags(mAccountData.getRoomTagsKeys());
                             getStore().flushSummary(summary);
```
Comment (BillCarsonFr) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: maybe you could flush a t the end of the for

Comment (giomfo) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: I agree, I will merge first PR #531 on this branch then apply your suggestion

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java [hunk 2]
```diff
@@ -1971,26 +1969,13 @@ private void handleRoomAccountDataEvents(List<Event> accountDataEvents) {
...
                         }
                         mDataHandler.onRoomTagEvent(getRoomId());
-                    } else if (Event.EVENT_TYPE_URL_PREVIEW.equals(accountDataEvent.getType())) {
```
[the remark this answers is not in the document]
Comment (giomfo) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: We still apply here some change in the room summary for `EVENT_TYPE_TAGS`.
The `EVENT_TYPE_URL_PREVIEW` event is totally handled now by the room account data (`mAccountData`).

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java (hunk 1 above)
Comment (giomfo) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: @BillCarsonFr  don't you think this flush was required before calling `mDataHandler` listeners (see `.onRoomTagEvent` and `onReadMarkerEvent`)?
I hesitate to move this flush. wdyt?

Comment (BillCarsonFr) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: yes, keep it like that. better not break anything

**Pull request merged**